updates snowflake prometheus exporter (with NULL fix) (#6929)
grafana/agent: CHANGELOG.md

- Fix a bug where a topic was claimed by the wrong consumer type in `otelcol.receiver.kafka`. (@wildum)

> - Updates Snowflake Prometheus Exporter dependency to [latest](https://github.com/grafana/snowflake-prometheus-exporter) as of May 28, 2024 (@StefanKurek)

Reviewer: ```suggestion
- Update `prometheus.exporter.snowflake` with the [latest](https://github.com/grafana/snowflake-prometheus-exporter) version of the exporter as of May 28, 2024 (@StefanKurek)
```
